### PR TITLE
1442 coverage upd

### PR DESCRIPTION
**Pull request by RamilGaripov**
Ticket: https://github.com/metriport/metriport-internal/issues/1442

### Description

- Slight improvement to Coverage, which allows for correct status mapping and does not erroneously remove coverages that don't have dates

### Testing

- Local
  - [x] Ran with a bunch of different XMLs and looked at MR outputs
  - [x] Tested with 10k files. Report is at the bottom

### Release Plan
- [ ] Merge this

### CODE REVIEW

File: packages/fhir-converter/src/templates/cda/Resources/Coverage.hbs [hunk 1]
```diff
@@ -9,10 +9,9 @@
             	{{>DataType/Identifier.hbs id=this}},
             {{/each}}
         ], 
-        "status":{{>ValueSet/FMStatus.hbs payerEntry.statusCode.code}},
+        "status":{{>ValueSet/FMStatus.hbs code=payerEntry.statusCode.code}},
```
Comment (jonahkaye) [inline packages/fhir-converter/src/templates/cda/Resources/Coverage.hbs:12]: good catch!

**Review by Goncharo — CHANGES_REQUESTED** [review]
@RamilGaripov changes LGTM, just a couple Qs:

- why are we leaving  `CCDA Policy Holder` & `Guarantor` as TODOs
- let's get in the habit of running these sorts of changes against our 10k CDA dataset - @jonahkaye can help with this

Once those are addressed, I'll give approval 👍

**Comment by RamilGaripov** [comment]
> @RamilGaripov changes LGTM, just a couple things:
> 
> * why are we leaving  `CCDA Policy Holder` & `Guarantor` as TODOs?
> * let's get in the habit of running these sorts of changes against our 10k CDA dataset - @jonahkaye can help with this
> 
> Once those are addressed, I'll give approval 👍

Conversation regarding `policyHolder and guarantor` are taken to Slack - https://metriport.slack.com/archives/C06CQC7GQG3/p1706816728255519

Test results (prior to latest fix that takes care of HAPI 1851)
```Converted 9919 files in 224681 ms. [4 errors]
>>> 459 files were skipped because they have nonXMLBody
Removing all partitions from the FHIR Server...
Creating tenant @ FHIRServer w/ {"cxId":"018d6678-06fe-706a-ad5a-6b73e63f13c5","organizationNumber":196350}
Found 9464 JSON files.
Inserting 9464 files, 1 at a time...


Error inserting runs/fhir-converter-e2e/2024-02-01T20:59:18.655Z/output/circle-medical-100-adhd-patient-sample/018a82ad-7298-7c02-9e58-e6b14a415610/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ad-7298-7c02-9e58-e6b14a415610_018ae4b1-e812-7db3-8ce7-df33d0afb9ca.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "1 - 2": Character   is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting /circle-medical-100-adhd-patient-sample/018a82ad-7298-7c02-9e58-e6b14a415610/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ad-7298-7c02-9e58-e6b14a415610_018ae4b1-e812-7db3-8ce7-df33d0afb9ca.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "1 - 2": Character   is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting runs/fhir-converter-e2e/2024-02-01T20:59:18.655Z/output/circle-medical-100-adhd-patient-sample/018a82ad-7298-7c02-9e58-e6b14a415610/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ad-7298-7c02-9e58-e6b14a415610_018ae4b1-e812-7db3-8ce7-df2465604a13.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "1 - 2": Character   is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting /circle-medical-100-adhd-patient-sample/018a82ad-7298-7c02-9e58-e6b14a415610/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ad-7298-7c02-9e58-e6b14a415610_018ae4b1-e812-7db3-8ce7-df2465604a13.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "1 - 2": Character   is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting runs/fhir-converter-e2e/2024-02-01T20:59:18.655Z/output/circle-medical-100-adhd-patient-sample/018a82ad-7298-7c02-9e58-e6b14a415610/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ad-7298-7c02-9e58-e6b14a415610_018ae4b1-e812-7db3-8ce7-df0be1c6187d.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "1 - 2": Character   is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting /circle-medical-100-adhd-patient-sample/018a82ad-7298-7c02-9e58-e6b14a415610/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ad-7298-7c02-9e58-e6b14a415610_018ae4b1-e812-7db3-8ce7-df0be1c6187d.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "1 - 2": Character   is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting runs/fhir-converter-e2e/2024-02-01T20:59:18.655Z/output/circle-medical-100-adhd-patient-sample/018a82ad-7298-7c02-9e58-e6b14a415610/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ad-7298-7c02-9e58-e6b14a415610_018ae4b1-e812-7db3-8ce7-defc41dcb6ab.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "1 - 2": Character   is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting /circle-medical-100-adhd-patient-sample/018a82ad-7298-7c02-9e58-e6b14a415610/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ad-7298-7c02-9e58-e6b14a415610_018ae4b1-e812-7db3-8ce7-defc41dcb6ab.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "1 - 2": Character   is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting runs/fhir-converter-e2e/2024-02-01T20:59:18.655Z/output/circle-medical-100-adhd-patient-sample/018a82ac-a126-7611-9f49-bd4eb872e20f/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ac-a126-7611-9f49-bd4eb872e20f_018a87bc-c0b7-7724-8754-5d516c06cfa0.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "&lt;&#x3D;200": Character & is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting /circle-medical-100-adhd-patient-sample/018a82ac-a126-7611-9f49-bd4eb872e20f/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ac-a126-7611-9f49-bd4eb872e20f_018a87bc-c0b7-7724-8754-5d516c06cfa0.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "&lt;&#x3D;200": Character & is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting runs/fhir-converter-e2e/2024-02-01T20:59:18.655Z/output/circle-medical-100-adhd-patient-sample/018a82ac-7ae8-7ab2-8509-85fe3b9ccf63/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ac-7ae8-7ab2-8509-85fe3b9ccf63_018ae4af-3f5c-7fb0-ad18-f4798ea964b4.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "&amp;lt;&#x3D;10": Character & is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting /circle-medical-100-adhd-patient-sample/018a82ac-7ae8-7ab2-8509-85fe3b9ccf63/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ac-7ae8-7ab2-8509-85fe3b9ccf63_018ae4af-3f5c-7fb0-ad18-f4798ea964b4.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "&amp;lt;&#x3D;10": Character & is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting runs/fhir-converter-e2e/2024-02-01T20:59:18.655Z/output/circle-medical-100-adhd-patient-sample/018a82ac-7ae8-7ab2-8509-85fe3b9ccf63/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ac-7ae8-7ab2-8509-85fe3b9ccf63_018a87e2-2892-72f8-816d-2d157a3af475.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "&amp;lt;&#x3D;10": Character & is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting /circle-medical-100-adhd-patient-sample/018a82ac-7ae8-7ab2-8509-85fe3b9ccf63/5af0e105-9439-4c02-939b-ecf7b230b418_018a82ac-7ae8-7ab2-8509-85fe3b9ccf63_018a87e2-2892-72f8-816d-2d157a3af475.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "&amp;lt;&#x3D;10": Character & is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting runs/fhir-converter-e2e/2024-02-01T20:59:18.655Z/output/error__full_cda.xml.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1838: Invalid JSON content detected, missing required element: 'resourceType'
Error inserting /error__full_cda.xml.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1838: Invalid JSON content detected, missing required element: 'resourceType'
Error inserting runs/fhir-converter-e2e/2024-02-01T20:59:18.655Z/output/error__circle-medical-100-adhd-patient-sample_018a82ae-5a33-7508-bc2c-1840a929680d_5af0e105-9439-4c02-939b-ecf7b230b418_018a82ae-5a33-7508-bc2c-1840a929680d_018ad290-61e9-711c-a240-f4a7b9fe955a.xml.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1838: Invalid JSON content detected, missing required element: 'resourceType'
Error inserting /error__circle-medical-100-adhd-patient-sample_018a82ae-5a33-7508-bc2c-1840a929680d_5af0e105-9439-4c02-939b-ecf7b230b418_018a82ae-5a33-7508-bc2c-1840a929680d_018ad290-61e9-711c-a240-f4a7b9fe955a.xml.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1838: Invalid JSON content detected, missing required element: 'resourceType'
Error inserting runs/fhir-converter-e2e/2024-02-01T20:59:18.655Z/output/error__circle-medical-100-adhd-patient-sample_018a82ae-5a33-7508-bc2c-1840a929680d_5af0e105-9439-4c02-939b-ecf7b230b418_018a82ae-5a33-7508-bc2c-1840a929680d_018ad290-61e9-711c-a240-f4a668fabfd5.xml.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1838: Invalid JSON content detected, missing required element: 'resourceType'
Error inserting /error__circle-medical-100-adhd-patient-sample_018a82ae-5a33-7508-bc2c-1840a929680d_5af0e105-9439-4c02-939b-ecf7b230b418_018a82ae-5a33-7508-bc2c-1840a929680d_018ad290-61e9-711c-a240-f4a668fabfd5.xml.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1838: Invalid JSON content detected, missing required element: 'resourceType'
Error inserting runs/fhir-converter-e2e/2024-02-01T20:59:18.655Z/output/error__circle-medical-100-adhd-patient-sample_018a82ad-0c80-7d03-878e-b0bbb742aa7d_5af0e105-9439-4c02-939b-ecf7b230b418_018a82ad-0c80-7d03-878e-b0bbb742aa7d_018ae4b0-9b99-740e-bd6f-043f27666011.xml.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1838: Invalid JSON content detected, missing required element: 'resourceType'
Error inserting /error__circle-medical-100-adhd-patient-sample_018a82ad-0c80-7d03-878e-b0bbb742aa7d_5af0e105-9439-4c02-939b-ecf7b230b418_018a82ad-0c80-7d03-878e-b0bbb742aa7d_018ae4b0-9b99-740e-bd6f-043f27666011.xml.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1838: Invalid JSON content detected, missing required element: 'resourceType'
Error inserting runs/fhir-converter-e2e/2024-02-01T20:59:18.655Z/output/big.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "&amp;lt;&#x3D;10": Character & is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting /big.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "&amp;lt;&#x3D;10": Character & is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting runs/fhir-converter-e2e/2024-02-01T20:59:18.655Z/output/10.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "&amp;lt;&#x3D;10": Character & is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Error inserting /10.json: HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element="value"] Invalid attribute value "&amp;lt;&#x3D;10": Character & is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
Inserted 9464 files in 2775349 ms. [13 errors]
Resources:  {
  AllergyIntolerance: 741,
  Communication: 790,
  Composition: 9420,
  Condition: 12898,
  Consent: 44,
  Coverage: 559,
  Device: 3834,
  DiagnosticReport: 21177,
  Encounter: 18459,
  FamilyMemberHistory: 724,
  Goal: 45,
  Immunization: 2087,
  MedicationAdministration: 3207,
  MedicationRequest: 3182,
  MedicationStatement: 4538,
  Observation: 77766,
  Procedure: 8803,
  RelatedPerson: 271,
  ServiceRequest: 1
}
Total resources: 168546
Total time: 2781274 ms / 46.35456666666666 min```

File: packages/fhir-converter/src/templates/cda/Sections/Payers.hbs [hunk 2]
```diff
@@ -13,10 +13,11 @@
 
                     Example: https://github.com/HL7/C-CDA-Examples/blob/fb1c33b3a880fc47407f25c5dafa0066996529f4/Documents/CCD/CCD%201/CCD.XML
                     --}}
-                    {{#if (contains (toJsonString this.templateId) '"2.16.840.1.113883.10.20.22.4.87"')}}
+                    {{#if (contains (toJsonString this.templateId) "2.16.840.1.113883.10.20.22.4.87")}}
                         {{#if this.assignedEntity.representedOrganization.name._}}
                             {{>Resources/Organization.hbs org=this.assignedEntity.representedOrganization ID=(generateUUID (toJsonString this.assignedEntity.representedOrganization))}},
-                            {{>References/Coverage/payor.hbs ID=(generateUUID (toJsonString payEntry.act)) REF=(concat 'Organization/' (generateUUID (toJsonString this.assignedEntity.representedOrganization)))}},
+                            {{>References/Coverage/payor.hbs ID=(generateUUID (toJsonString payerEntry.act)) REF=(concat 'Organization/' (generateUUID (toJsonString this.assignedEntity.representedOrganization)))}},
```
Comment (jonahkaye) [inline packages/fhir-converter/src/templates/cda/Sections/Payers.hbs:19]: good well done example of new pattern: 
always generate references to Organizations / Providers etc alongside actually creating the Organization or Provider. This way we dont ever have dangling references!

**Pull request closed without merging**